cluster.c

```diff
@@ -379,6 +379,58 @@ RRStatus ShardGroupAppendLogEntry(RedisRaftCtx *rr, ShardGroup *sg, int type, vo
     return RR_OK;
 }
 
+RRStatus ShardGroupsAppendLogEntry(RedisRaftCtx *rr, int num_sg, ShardGroup **sg, int type, void *user_data)
+{
+    /* Make sure we're still a leader, could have changed... */
+    /* SJP: unsure this is true in this case */
+    if (!raft_is_leader(rr->raft)) {
```

Comment:
There is a checkLeader() call before calling this function. So, I believe you can remove this check here.

Reply:
so I realized that when writing, but my concern was that perhaps this function (much like add) might be called in the future from within rediraft itself and it miss the check.  If we dont want to worry about that now, easy enough to remove.

Reply:
Even then, probably this check is not necessary. There is `raft_recv_entry` call inside this function, it will/should fail anyway if this node is not the leader.

Reply:
ok, removed